lms/djangoapps/instructor_task/tasks_helper.py

```diff
@@ -588,6 +589,10 @@ def delete_problem_module_state(xmodule_instance_args, _module_descriptor, stude
     Always returns UPDATE_STATUS_SUCCEEDED, indicating success, if it doesn't raise an exception due to database error.
     """
     student_module.delete()
+
+    # Remove milestones that user has completed
+    milestones_helpers.remove_course_milestone(student_module.course_id, student_module.student)
```

Comment:
I'm not sure that we want to remove ALL milestones for this course+user combination in the context of deleting the state for a particular student problem module.  We should probably add a remove_course_content_milestone method to milestones_helpers and use that instead.